tests/test_optimizely_factory.py

```diff
@@ -0,0 +1,168 @@
+# Copyright 2021, Optimizely
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import mock
+
+from optimizely.config_manager import PollingConfigManager
+from optimizely.error_handler import NoOpErrorHandler
+from optimizely.event_dispatcher import EventDispatcher
+from optimizely.helpers import validator
+from optimizely.notification_center import NotificationCenter
+from optimizely.optimizely_factory import OptimizelyFactory
+from optimizely.user_profile import UserProfileService
+from . import base
+
+
+@mock.patch('requests.get')
+class OptimizelyFactoryTest(base.BaseTest):
+    def setUp(self):
+        self.datafile = '{ revision: "42" }'
+        self.error_handler = NoOpErrorHandler()
+        self.mock_client_logger = mock.MagicMock()
+        self.notification_center = NotificationCenter(self.mock_client_logger)
+        self.event_dispatcher = EventDispatcher()
+        self.user_profile_service = UserProfileService()
+
+    def test_default_instance_should_create_config_manager_when_sdk_key_is_given(self, _):
```

Comment:
please use double underscore in test name after the main part of the test name. In All tests. For example:
1. test_default_instance__should_create_config_manager_when_sdk_key_is_given
2. test_default_instance__should_create_config_set_default_values_when_polling_int_and_blocking_time_invalid
3. etc

This is for consistency with other test names. Double underscore clarifies the specifics of the test in the second part of the test name, so it's nice to have that double underscore mark.